src/olympia/devhub/forms.py

```diff
@@ -586,10 +586,11 @@ def clean(self):
             self._clean_upload()
             xpi = parse_addon(self.cleaned_data['upload'], self.addon)
             # Make sure we don't already have the same non-rejected version.
-            if self.addon.versions.filter(version=xpi['version']).exclude(
-                    files__status=amo.STATUS_DISABLED):
+            if Version.unfiltered.filter(addon=self.addon,
+                                         version=xpi['version']):
```

Comment:
The result is always going to be 0 or 1 results, fwiw (unless something has gone horribly wrong).